contracts/lease/src/contract/state/controller.rs

```diff
@@ -130,9 +130,9 @@ pub fn execute(
                  cw_response,
                  next_state,
              }| {
-                super::save(deps.storage, &next_state)?;
-
-                Ok(cw_response)
+                super::save(deps.storage, &next_state)
+                    .map(move |()| cw_response)
+                    .map_err(Into::into)
```

Comment:
two arguments against applying the functional style here:
- only this occurrence in this module changed, there are at least a few more
- the previous code was shorter and easier to read